tools/user_migration/migrate_user.py

```diff
@@ -0,0 +1,374 @@
+#! venv/bin/python
+
+# painfully stupid script to check out dumping mysql databases to avro.
+# Avro is basically "JSON" for databases. It's not super complicated & it has
+# issues (one of which is that it requires Python2).
+#
+#
+
+import argparse
+import logging
+import base64
+import sys
+import time
+from datetime import datetime
+
+from mysql import connector
+from google.cloud import spanner
+from google.api_core.exceptions import AlreadyExists
+from urllib.parse import urlparse
+
+SPANNER_NODE_ID = 800
+
+
+class BadDSNException(Exception):
+    pass
+
+
+class Collections:
+    """Cache spanner collection list.
+
+    The spanner collection list is the (soon to be) single source of
+    truth regarding collection ids.
+
+    """
+    _by_name = {}
+    databases = None
+
+    def __init__(self, databases):
+        """Get the cache list of collection ids"""
+        sql = """
+        SELECT
+            name, collection_id
+        FROM
+            collections;
+        """
+        self.databases = databases
+        logging.debug("Fetching collections...")
+        with self.databases['spanner'].snapshot() as cursor:
+            rows = cursor.execute_sql(sql)
+            for row in rows:
+                self._by_name[row[0]] = row[1]
+
+    def get_id(self, name):
+        """ Get/Init the ID for a given collection """
+        if name in self._by_name:
+            return self._by_name.get(name)
+
+        # option 1: do what spanner-rs does
+        with self.databases['spanner'].batch() as cursor:
+            result = cursor.execute_sql("""
+                SELECT
+                    COALESCE(MAX(collection_id), 1)
+                FROM
+                    collections""")
+            collection_id = result.one()[0] + 1
```

Comment:
I forgot this:
```suggestion
            collection_id = max(result.one()[0] + 1, 101)
```

Reply:
Not sure about this. I'm guessing that we're going to pre-populate the `collections` table with the knowns. I'm guessing we'd also include a `100:reserved` marker so that the custom values start above that.

Granted, the `collection_id` is only used by the server. I'm not really sure what value the reserved range really is, but that's just my ignorance.

Reply:
The standard/reserved collections we hardcode should have been pre-populated in the collections table (but we should really double check that this did indeed happen on prod first).

I'm leaning towards implementing this max line for now, to match the syncstorage implementations. But we should probably have a conversation with rfk about how this reservation's done and what we want to do going forward